lib/blpapi-wrapper.ts

```diff
@@ -320,12 +320,10 @@ export class Session extends events.EventEmitter {
         this.openService(uri).then((): void => {
             log(util.format('Request: %s|%d', requestName, correlatorId));
             trace(request);
-            if (!(requestName in REQUEST_TO_RESPONSE_MAP)) {
-                throw new Error('Unknown request type ' + requestName);
-            }
-            var responseEventName = REQUEST_TO_RESPONSE_MAP[requestName];
             this.session.request(uri, requestName, request, correlatorId);
-            this.listen(responseEventName,
+            assert(requestName in REQUEST_TO_RESPONSE_MAP,
+                   util.format('Request, %s, not handled', requestName));
```

Comment:
Isn't the requestName provided by the user? Is that the case we shouldn't assert if the lookup fails.